### PR TITLE
feat(new): initial module for Private DNS Zone

### LINKED ISSUE #382: [Module Proposal]: Private DNS Zones
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

Private DNS Zone: To provision Private DNS Zone using BICEP module with different parameters.
https://learn.microsoft.com/en-us/azure/templates/microsoft.network/privatednszones?pivots=deployment-language-bicep

### Module path

network/private-dns-zone

### Describe the module

Private DNS Zone: To provision Private DNS Zone using BICEP module with different parameters.
https://learn.microsoft.com/en-us/azure/templates/microsoft.network/privatednszones?pivots=deployment-language-bicep

### PR DESCRIPTION
**Pull request by schandnani**
## Description

Fixes #382 

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Worksflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Comment by schandnani** [comment]
hello @dciborow pls review, thanks!

**Review by shenglol — APPROVED** [review]

File: modules/network/private-dns-zone/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+| :--- | :----: | :---------------------- |
+| id   | string | Get privateDnsZone ID   |
+| name | string | Get privateDnsZone name |
+
+## Examples
+
+### Example 1
+
+```
+module dnsZone 'br/public:network/private-dns-zone:*' = {
```
Comment (shenglol) [inline modules/network/private-dns-zone/README.md:32]:
```suggestion
module dnsZone 'br/public:network/private-dns-zone:1.0.1' = {
```

**Review by shenglol — CHANGES_REQUESTED** [review]
Looks good. Just one suggestion for upadating the module version in the example.

**Comment by schandnani** [comment]
@shenglol done!!

File: modules/network/private-dns-zone/README.md (hunk 1 above)
Comment (schandnani) [reply modules/network/private-dns-zone/README.md]: done

File: modules/network/private-dns-zone/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+| :----- | :------: | :---------------------- |
+| `id`   | `string` | Get privateDnsZone ID   |
+| `name` | `string` | Get privateDnsZone name |
+
+## Examples
+
+### Example 1
+
+```
+module dnsZone 'br/public:network/private-dns-zone:0.1.0' = {
```
Comment (shenglol) [inline modules/network/private-dns-zone/README.md:32]:
```suggestion
module dnsZone 'br/public:network/private-dns-zone:1.0.1' = {
```
Looks like this was changed to `1.0.1` which is correct but got changed to `0.1.0` later?

Comment (schandnani) [reply modules/network/private-dns-zone/README.md]: Now it is correct 0.0.1

Comment (shenglol) [reply modules/network/private-dns-zone/README.md]: Sorry, I mean we should change it to `1.0.1`. For any new module, the version number for the examples should be whatever the version number defined in `version.json` (usually `1.0`) + `.1`. In this case the major and minor version in `version.json` is `1.0`, so the full version should be `1.0.1` once the PR is merged.

Comment (schandnani) [reply modules/network/private-dns-zone/README.md]: Can you please review now

**Review by shenglol — APPROVED** [review]

**Pull request merged**